### PR TITLE
Bug 1183278-Too easy to add more than 5 series in graphs UI

### PR DESCRIPTION
**Pull request by MikeLing**
One thing must been mentioned that I still using 2 indentations instead of 4, just keep synchronize with other code. Please tell me if I should use 4 in this PR.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/770)

### CODE REVIEW

File: ui/js/graphs.js [hunk 1]
```diff
@@ -688,6 +688,9 @@ perf.controller('GraphsCtrl', [
                 timeRange: function() {
                   return $scope.myTimerange.value;
                 },
+                testCounter: function() {
```
Comment (wlach) [inline ui/js/graphs.js]: I don't think this needs to be a function, you can just set it to the value of $scope.seriesList.length.

Let's change the name from `testCounter` to `numTestsDisplayed`. More descriptive.

File: ui/js/graphs.js [hunk 2]
```diff
@@ -738,6 +741,24 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
...
+        addTestToGraph();
+      }
+    } else if ($scope.addedTestList.length + testCounter > 6) {
+      var a = window.confirm('WARNING: You have ' + testCounter + ' tests in the graph already' +
+                          ' and You are about to load ' + $scope.addedTestList.length + ' more test.\n'
+                           + 'Loading more than 6 is not recommended. Do it anyway?');
+      if (a == true) {
+        addTestToGraph();
+      }
+    } else addTestToGraph();
```
Comment (wlach) [inline ui/js/graphs.js]: There's no need for two if..else clauses here. The second condition is a superset of the first. :)

i.e. all you need is:

    if ( $scope.addedTestList.length + testCounter > 6) {
        ...
    } else {
      addTestToGraph();
    }

File: ui/js/graphs.js [hunk 3]
```diff
@@ -738,6 +741,24 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
...
+    if ($scope.addedTestList.length > 6) {
+      var a = window.confirm('WARNING: You are about to load ' + $scope.addedTestList.length +
+                           ' tests to the graph.\n' +
+                           'Loading more than 6 is not recommended. Do it anyway?');
+      if (a == true) {
+        addTestToGraph();
+      }
+    } else if ($scope.addedTestList.length + testCounter > 6) {
+      var a = window.confirm('WARNING: You have ' + testCounter + ' tests in the graph already' +
+                          ' and You are about to load ' + $scope.addedTestList.length + ' more test.\n'
```
Comment (wlach) [inline ui/js/graphs.js]: A few problems:

* The capitalization is wrong here ("and You are...").
* tests should be plural at the end ("you are about to load X more tests").

I don't think we need to spell things out in so much detail for the user. It should be pretty obvious what's going on already. Let's just say something simple like 'WARNING: Displaying more than 6 graphs at the same time is not recommended. Do it anyway?' That way you can implement the simplification I suggest below.

File: ui/partials/perf/testdatachooser.html [hunk 4]
```diff
@@ -44,5 +44,5 @@ <h3 class="modal-title">Add test data</h3>
   </div>
 </div>
 <div class="modal-footer">
-  <button class="btn btn-primary" ng-click="addTestData()" ng-disabled="!addedTestList.length" ng-model="rightList"><span class="glyphicon glyphicon-plus" aria-hidden="true"></span> Add</button>
+  <button class="btn btn-primary" ng-click="addTestData()" ng-disabled="!addedTestList.length" ng-model="rightList"><span class="glyphicon glyphicon-plus" aria-hidden="true"></span> Add ({{addedTestList.length}})</button>
```
Comment (wlach) [inline ui/partials/perf/testdatachooser.html]: This isn't necessary, we can leave it out. It's already obvious how many tests are going to be added from glancing at the list. :)

File: ui/js/graphs.js (hunk 2 above)
Comment (wlach) [inline ui/js/graphs.js]: I just realized that you wanted to display two different messages depending on the situation. But IMO this isn't necessary -- we can simplify the code and the display of information by using a shorter message.

Comment (MikeLing) [inline ui/js/graphs.js]: Is that mean we don't need to consider how much tests been putted in graph already? We just need to tell user don't put more than 6 tests at once.

File: ui/js/graphs.js (hunk 1 above)
Comment (MikeLing) [inline ui/js/graphs.js]: yep, I just try it with just set the value of $scope.seriesList.length, but it tells me the equation can't be recognized. But I had changed the name of testCounter to numTestsDisplayed :)

File: ui/js/graphs.js (hunk 2 above)
Comment (wlach) [inline ui/js/graphs.js]: We do want to consider it in our calculation. There's just no need to display the exact counts in our message.

File: ui/js/graphs.js (hunk 1 above)
Comment (wlach) [inline ui/js/graphs.js]: Cool :)

File: ui/js/graphs.js (hunk 2 above)
Comment (MikeLing) [inline ui/js/graphs.js]: Oh, I see :)

File: ui/js/graphs.js [hunk 5]
```diff
@@ -738,6 +741,15 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
   var testArray = [];
   var series = [];
   $scope.addTestData = function () {
+    if ($scope.addedTestList.length + numTestsDisplayed > 6) {
+      var a = window.confirm('WARNING: Displaying more than 6 graphs at the same time is not recommended. Do it anyway?');
+      if (a == true) {
+        addTestToGraph();
+      }
+    } else addTestToGraph();
```
Comment (wlach) [inline ui/js/graphs.js]: Oh, forgot to mention, you should wrap the else statement with braces for clarity here.

i.e.

    else {
      addTestToGraph();
    }

**Comment by MikeLing** [comment]
screenshot from 2015-07-16 11 29 08

It looks like this one, now :)

**Comment by tojonmz** [comment]
Drive by comment :) I wonder if it would be helpful to explain why instead in that message. eg.

`...Displaying more than 5 graphs (something*). Do it anyway?`

where something is whatever the risk is, eg.
`can produce dense point results.`
`may be difficult to view.`
`may be slow to load.`
...or whatever the reason might be :)

**Comment by wlach** [comment]
Hmm, so there are several problems with displaying more than 6:

* our legend only supports 6 graphs
* that much data is pretty impossible to decipher

I'm not sure how to really describe that succinctly. I'm inclined to just merge this as-is, I don't think it's that important to explain this in detail. But if you have some wording that can express that succinctly I am open to suggestions. :)

**Comment by tojonmz** [comment]
My only attempt then, might be to reference the legend issue:

`Displaying more than 6 graphs is not supported in the UI. Do it anyway?`

But if the UI result produced is not useful, I wonder if we should allow the user to get in that state to begin with? Though whatever you guys think is fine with me :)

**Comment by wlach** [comment]
"why allow it at all" is a reasonable question :) I think mostly because I think restricting it in a reasonable and friendly way would be too much work at this juncture. I like your wording. I'll fix that up and then push this.

**Comment by wlach** [comment]
Merged manually: https://github.com/mozilla/treeherder/commit/b34690490a106efececd72dda9c9ecf2014f0b41

**Pull request closed without merging**